text/trace/0168-sampling-propagation.md

```diff
@@ -0,0 +1,368 @@
+
+Using six bits of information we can convey known and unknown sampling
+rates as small as 2**-61.  The value 63 is reserved to mean sampling
+with probability 0, which conveys an adjusted count of 0 for the
+associated context.
+
+When propagated, the probability value will be interpreted as shown in
+the folowing table, which uses an offset of +1 in order to place the
+Unknown value at 0:
+
+| Probability Value | Head Probability | Note                   |
+| -----             | -----------      | ----                   |
+| 0                 | Unknown          | Reserved for span data |
+| 1                 | 1                |                        |
+| 2                 | 1/2              |                        |
+| 3                 | 1/4              |                        |
+| ...               | ...              |                        |
+| N                 | 2**(-N+1)        | 1 in 2**(N-1)          |
+| ...               | ...              |                        |
+| 61                | 2**-60           |                        |
+| 62                | 2**-61           |                        |
+| 63                | 0                | Maximum encoded value  |
+
+[Described in OTEP
+170](https://github.com/open-telemetry/oteps/pull/170), Span data
+sampled by the `ParentBased` sampler will encode the value that was
+propagated by the parent span as its "probability value" `p`.
+
+The value `p=0` SHOULD NOT be propagated using `tracestate`
+explicitly, because the equivalent interpretation can be obtained by
+omitting `p`.
+
+### Randomness value
+
+With head trace sampling probabilities limited to powers of two, the
+amount of randomness needed per trace context is limited.  A
+consistent sampling decision is accomplished by propagating a specific
+random variable denoted `r`.  The random variable is a described by a
+geometric distribution having shape parameter `1/2`, listed below:
+
+| `r` Value | Selection Probability |
+| ---------------- | --------------------- |
+| 0 | 1/2 |
+| 1 | 1/4 |
+| 2 | 1/8 |
+| 3 | 1/16 |
+| ... | ... |
+| 0 <= `r` <= 61 | 1/(2**(-`r`+1)) |
+| ... | ... |
+| 60 | 2**-61 |
+| 61 | 2**-61 |
+
+Such a random variable `r` can be generated using the following
+pseudocode.
+
+```golang
+func nextRandomness() int {
+  r := 0
+  for r < 61 && nextRandomBit() == false {
+    r++
+  }
+  return R
+}
+```
+
+This can be computed from a stream of random bits as the number of
+leadieng zeros using efficient instructions on modern computer
+architectures.
+
+For example, the value 3 means there were three leading zeros and
+corresponds with being sampled at probabilities 1-in-1 through 1-in-8
+but not at probabilities 1-in-16 and smaller.
+
+### Proposed `tracestate` syntax
+
+The consistent sampling randomness valuw (`r`) and and head sampling
+probability value (`p`) will be propagated using two bytes of base16 content
+for each of the two fields, as follows,
+
+```
+tracestate: otel=p:PP;r:RR
```

Comment:
tracestate key value pairs might get [removed based on FIFO](https://www.w3.org/TR/trace-context/#tracestate-limits), do we have recommendation for OpenTelemetry SDKs such as "the OpenTelemetry SDKs are encouraged to preserve otel tracestate as much as possible, for example, always pop it to 1st key-value pair of tracestate" (which might be a bit off from the W3C TraceContext perspective).